changelog/8724.feature.md

```diff
@@ -0,0 +1 @@
+Added `sasl_mechanism` as an optional configurable parameters for the [Kafka Producer](https://kafka-python.readthedocs.io/en/master/apidoc/KafkaProducer.html).
```

Comment:
Shouldn't the link go to the Rasa docs? 🤔

Reply:
Yep 🤦